src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -1,41 +1,127 @@
 package io.stargate.sgv2.jsonapi.service.operation.model.impl;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import io.smallrye.mutiny.Uni;
 import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.sgv2.api.common.schema.SchemaManager;
 import io.stargate.sgv2.jsonapi.api.model.command.CommandContext;
 import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import io.stargate.sgv2.jsonapi.service.bridge.executor.CollectionSettings;
 import io.stargate.sgv2.jsonapi.service.bridge.executor.QueryExecutor;
 import io.stargate.sgv2.jsonapi.service.operation.model.Operation;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 import java.util.function.Supplier;
 
 public record CreateCollectionOperation(
     CommandContext commandContext,
+    ObjectMapper objectMapper,
+    SchemaManager schemaManager,
     String name,
     boolean vectorSearch,
     int vectorSize,
     String vectorFunction,
     String vectorize)
     implements Operation {
 
+  private static final Function<String, Uni<? extends Schema.CqlKeyspaceDescribe>>
+      MISSING_KEYSPACE_FUNCTION =
+          keyspace -> {
+            String message =
+                "Unknown namespace '%s', you must create it first.".formatted(keyspace);
+            Exception exception = new JsonApiException(ErrorCode.NAMESPACE_DOES_NOT_EXIST, message);
+            return Uni.createFrom().failure(exception);
+          };
+
   public static CreateCollectionOperation withVectorSearch(
       CommandContext commandContext,
+      ObjectMapper objectMapper,
+      SchemaManager schemaManager,
       String name,
       int vectorSize,
       String vectorFunction,
       String vectorize) {
     return new CreateCollectionOperation(
-        commandContext, name, true, vectorSize, vectorFunction, vectorize);
+        commandContext,
+        objectMapper,
+        schemaManager,
+        name,
+        true,
+        vectorSize,
+        vectorFunction,
+        vectorize);
   }
 
   public static CreateCollectionOperation withoutVectorSearch(
-      CommandContext commandContext, String name) {
-    return new CreateCollectionOperation(commandContext, name, false, 0, null, null);
+      CommandContext commandContext,
+      ObjectMapper objectMapper,
+      SchemaManager schemaManager,
+      String name) {
+    return new CreateCollectionOperation(
+        commandContext, objectMapper, schemaManager, name, false, 0, null, null);
   }
 
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
+    return schemaManager
+        .getTable(commandContext.namespace(), name, MISSING_KEYSPACE_FUNCTION)
+        .onItemOrFailure()
```

Comment:
Maybe it would be better to separate `onItem` and `onFailure` cases? Failure can never proceed with actual operation, only needs to surface failure in question.

Reply:
Based on our discussion, this is necessary to work around problem with a unit test, mocking.